### PR TITLE
[WIP] Some updates to TX signing flows

### PR DESCRIPTION
**Pull request by hstove**
Not totally working yet, but publishing so Thomas can take a look while I work on other things.

### CODE REVIEW

File: packages/app/src/pages/transaction/index.tsx [hunk 1]
```diff
@@ -96,57 +132,55 @@ export const Transaction: React.FC = () => {
...
+  const setupWithState = async (tx: ContractCallPayload | ContractDeployPayload) => {
+    if (tx.txType === 'contract-call') {
+      const contractSource = await client.fetchContractSource({
+        contractName: tx.contractName,
+        contractAddress: tx.contractAddress,
+      });
+      setContractSrc(contractSource);
+      setPendingTransaction(tx);
+    }
+    if (tx.txType === 'contract-deploy') {
```
Comment (kyranjamie) [inline packages/app/src/pages/transaction/index.tsx:144]: In every other instance I've seen, tx types are denoted as [`contract_call`](https://github.com/blockstack/stacks-blockchain-sidecar/blob/master/docs/entities/transactions/transaction-2-contract-call.schema.json#L16). Where does this value come from that is hyphenated?

Comment (kyranjamie) [reply packages/app/src/pages/transaction/index.tsx:144]: A string literal type, `TransactionType` is available in the types package, v0.0.13: https://github.com/blockstack/stacks-blockchain-sidecar/pull/73, if you can use this

Comment (aulneau) [reply packages/app/src/pages/transaction/index.tsx:144]: perfect, thank you. I noticed it has been named differently in a few places. Very helpful!

**Comment by aulneau** [comment]
closing in favor of https://github.com/blockstack/ux/pull/367

**Pull request closed without merging**